src/middlewared/middlewared/plugins/pool.py

```diff
@@ -324,6 +324,14 @@ class Config:
         )
     )
 
+    @accepts(Str('name'))
+    @returns(Ref('pool_entry'))
+    async def get_instance_by_name(self, name):
```

Comment:
1. Where is this called?
2. I also thought this would raise `IndexError` if `name` doesn't exist?

Reply:
I needed it, then I did not need it anymore, but I still think it might be useful in the future.

Reply:
Will it raise a `ValidationError` though like the docstring says it will? I thought it will be `IndexError`?

Reply:
I pushed a commit that fixes this

Reply:
It raises `InstanceNotFound` which is a validation error